packages/ntcore-ts-client/src/lib/pubsub/pubsub.ts

```diff
@@ -95,12 +133,17 @@ export class PubSubClient {
    * @param params - The announce message parameters.
    */
   private onTopicAnnounce = (params: AnnounceMessageParams) => {
-    const topic = this.topics.get(params.name);
-    if (!topic) {
+    let topic = this.topics.get(params.name);
+
+    if (!topic && this.onTopicAnnouncedListeners.length > 0) {
```

Comment:
This is a great step in the right direction! However, we need to consider that a topic could be announced before the user instantiates a listener. Could we instead create a list of all topic names that have been announced? That way, we could easily subscribe to all current and future topics regardless of the timing.

Possible implementations would be an array, set (hashset) or a map... I think a `Set` is probably the best? Let me know what you think

Reply:
Yeah, we should definitely trigger the callback with the previously announced topics!

I think you already have a source of truth of which topics are announced, since the topics already have an `announced` property.

So you could just go through the values of your topics map and filter for announced topics, like so:
```typescript
[...this.topics.values()].filter(topic => topic.announced).forEach(topic => callback(topic));
```

But if you'd prefer a dedicated data structure in pubsub, yeah I could add a `Set`.